src/main/java/org/jenkinsci/plugins/casc/JenkinsConfigurator.java

```diff
@@ -73,10 +73,11 @@ public void setValue(Jenkins jenkins, Object value) throws Exception {
             public void setValue(Jenkins jenkins, Object value) throws Exception {
                 List<Cloud> clouds = (List<Cloud>) value;
                 for (Cloud cloud : clouds) {
-                    if (jenkins.getCloud(cloud.name) == null) {
+                    Cloud currentCloud = jenkins.getCloud(cloud.name);
+                    if (currentCloud == null) {
                         jenkins.clouds.add(cloud);
                     } else {
-                        // FIXME re-configure ? remove/replace ?
+                        jenkins.clouds.replace(currentCloud, cloud);
```

Comment:
I wonder we could generalize this and introduce some `DescribableListAttribute` type for this use-case.